src/ModelContextProtocol/Server/McpServerExtensions.cs

```diff
@@ -210,4 +222,43 @@ async IAsyncEnumerable<ChatResponseUpdate> IChatClient.GetStreamingResponseAsync
         /// <inheritdoc/>
         void IDisposable.Dispose() { } // nop
     }
+
+    /// <summary>
+    /// Provides an <see cref="ILogger"/> implementation for sending logging message notifications
+    /// to the client for logged messages.
+    /// </summary>
+    private sealed class ClientLogger(IMcpServer server) : ILogger
```

Comment:
```suggestion
    private sealed class ClientLogger(IMcpServer server) : ILoggerProvider
```

Would it make more sense to make this an ILoggerProvider?

That way, consumers can control the category names of the logs they emit, but more importantly, it's much easier to plug into hosting by either registering the ILoggerProvider as a singleton with DI which will get picked up by the default LoggerFactory's constructor, or adding after the host is built via ILoggerFactory.AddProvider().